addons/mass_mailing_sms/wizard/mailing_sms_test.py

```diff
@@ -12,28 +12,51 @@ class MassSMSTest(models.TransientModel):
     def _default_numbers(self):
         return self.env.user.partner_id.phone_sanitized or ""
 
-    numbers = fields.Char(string='Number(s)', required=True,
-                          default=_default_numbers, help='Comma-separated list of phone numbers')
+    numbers = fields.Text(string='Number(s)', required=True,
+                          default=_default_numbers, help='Carriage-return-separated list of phone numbers')
     mailing_id = fields.Many2one('mailing.mailing', string='Mailing', required=True, ondelete='cascade')
 
     def action_send_sms(self):
         self.ensure_one()
-        numbers = [number.strip() for number in self.numbers.split(',')]
+
+        notification_messages = []
+        numbers = [number.strip() for number in self.numbers.split('\n')]
         sanitize_res = phone_validation.phone_sanitize_numbers_w_record(numbers, self.env.user)
-        sanitized_numbers = [info['sanitized'] for info in sanitize_res.values() if info['sanitized']]
+        sanitized_numbers = [info['sanitized'] for info in sanitize_res.values() if info['sanitized'] and not info['code']]
```

Comment:
Mmmh ? what is this code encore ?

Reply:
It's used to remove invalid numbers from the sanitized ones (see "invalid_numbers" below).

I was not necessary before since we raised if any info returned with a "code".

This "code" is returned from the phone_validation tool and can be either False, "invalid" or "empty".
(We don't make a difference between invalid and empty here, as it does not really matter).

Reply:
How do you get something in sanitized AND a code ? phone_sanitize_numbers does not seem to do that ...